common/src/state/mod.rs

```diff
@@ -500,9 +501,9 @@ impl State {
                     if let Some(msg) = chat
                         .messages
                         .iter_mut()
-                        .find(|msg| msg.inner.id() == message.inner.id())
+                        .find(|msg| msg.read().inner.id() == message.inner.id())
```

Comment:
I worry about `Signal::read()` here because the warp events are processed near the top of the VDOM. `process_warp_event()` is called by `use_app_coroutines()` which is called on line 169 of `main.rs`, in `app_layout`. 

After reading the comments in the [dioxus repository](https://github.com/DioxusLabs/dioxus/tree/master/packages/signals) it loos like `Signal::read()` automatically subscribes the caller's context. 

I suspect that storing `Signal`s in state isn't what we want. I could be wrong, but it seems like a `Signal` is better if used in a view rather than in a model.

Reply:
Signals will only subscribe inside of a component or a selector. Inside of async futures spawned in Dioxus' runtime (like in a use_coroutine future), or event handlers, they will not cause any new subscriptions

Reply:
That’s good news